### DISCUSSION TITLE
Can we use the new controlnets for FLUX without the need of touching the spaghetti backend?

### DISCUSSION THREAD

**Discussion by ZeroCool22** [mcmonkeyprojects/SwarmUI#243] [Q&A] [answered]
Thx in advance.

**Comment by mcmonkey4eva** [comment, marked as answer]
Most should *just work*, including [XLabs](https://huggingface.co/XLabs-AI/flux-controlnet-collections) and [InstantX](https://huggingface.co/InstantX/FLUX.1-dev-Controlnet-Canny) and [InstantX Union](https://huggingface.co/InstantX/FLUX.1-dev-Controlnet-Union)

just, yknow, name the files coherently (a lot of them are uploaded with terrible names like 'pytorch model', wtf).

There's a lot of R&D going on, so some controlnets might exist out there with their own unique formats that may not be supported

**Reply by ZeroCool22** [reply]
> Most should _just work_, including [XLabs](https://huggingface.co/XLabs-AI/flux-controlnet-collections) and [InstantX](https://huggingface.co/InstantX/FLUX.1-dev-Controlnet-Canny) and [InstantX Union](https://huggingface.co/InstantX/FLUX.1-dev-Controlnet-Union)
> 
> just, yknow, name the files coherently (a lot of them are uploaded with terrible names like 'pytorch model', wtf).
> 
> There's a lot of R&D going on, so some controlnets might exist out there with their own unique formats that may not be supported

Thx for your reply.

One last thing, i have made some **Symlinks** so i don't need to copy all models and other things to the ComfyUI folders.

If i update, those Symlinks will be broken and will need to create them again?

**Reply by mcmonkey4eva** [reply]
just set the model paths in Swarm's Server Configuration tab, you can give any path there and it will be applied to both swarm itself and the comfy backend automatically
